src/roswire/ros2/reader.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+import json
+import os
+import shlex
+import typing
+from typing import Any, Dict, Optional, Sequence, Tuple
+
+import attr
+import pkg_resources
+from loguru import logger
+
+from ..proxy.roslaunch.config import ExecutableType, LaunchConfig, NodeConfig
+from ..proxy.roslaunch.context import LaunchContext
+from ..proxy.roslaunch.reader import LaunchFileReader
+
+if typing.TYPE_CHECKING:
+    from .. import AppInstance
+
+
+@attr.s(auto_attribs=True)
+class ROS2LaunchFileReader(LaunchFileReader):
+
+    _app_instance: 'AppInstance'
+
+    def locate_node_binary(self, package: str, node_type: str) -> str:
+        raise NotImplementedError
+
+    def read(self,
+             fn: str,
```

Comment:
This breaks the expected interface for `read`.

Reply:
I think you're looking at LaunchFileReader and expecting LaunchManager